mods/core/src/common/auth/auth_middleware.ts

```diff
@@ -0,0 +1,53 @@
+import grpc from 'grpc'
+import Auth, { Jwt } from '@fonos/auth'
+import role_has_access from '../../../src/server/usermanager/role_has_access'
+export default class AuthMiddleware {
+  secretKeyToken: string
+  constructor (secretKey: string) {
+    this.secretKeyToken = secretKey
+  }
+
+  middleware = async (ctx: any, next: any, errorCb: any) => {
```

Comment:
You might simplify by importing the `errors` instead of passing this as a parameter.

> You will have to add a new error type.